app/src/org/commcare/android/framework/CommCareActivity.java

```diff
@@ -303,9 +304,8 @@ protected int getWakeLockLevel() {
      * occurred while the activity was paused.
      */
     private void syncTaskBlockingWithDialogFragment() {
-        if (dialogId < 0) {

```

Comment:
Can't remember why I decided to go this route.
I like your fix! 
_I think_ it is necessary to do this sort of check in case the activity is paused but the task finishes or is canceled.